cmd/kube-apiserver/app/options/options.go

```diff
@@ -87,6 +87,8 @@ func NewServerRunOptions() *ServerRunOptions {
 		},
 		ServiceNodePortRange: DefaultServiceNodePortRange,
 	}
+	// Overwrite the default for storage data format.
+	s.GenericServerRunOptions.DefaultStorageMediaType = "application/vnd.kubernetes.protobuf"
```

Comment:
Should this only default this way if etcd3 is chosen? If etcd2 is selected, does this default back to JSON?